{{cookiecutter.project_slug}}/Makefile

```diff
@@ -20,9 +20,9 @@ build:
 
 ## Recompile requirements and store pinned dependencies with hashes.
 pip-compile:
-	docker run --rm -v `pwd`/requirements:/build dddecaf/wsgi-base:alpine \
-		pip-compile --generate-hashes --output-file /build/requirements.txt \
-		/build/requirements.in
+	docker run --rm -v `pwd`/requirements:/build dddecaf/wsgi-base:compiler \
+		pip-compile --generate-hashes --upgrade \
```

Comment:
I'm wondering a bit about the upgrade. We'd want to upgrade the versions of what's in `requirements.in` but not what's coming from `base-requirements.in`, right?

Reply:
Yes, we're always constrained with the base. This is relevant when something in the base has upgraded. Without `--upgrade`, `pip-compile` would keep the version in the existing final `requirements.txt`. With it, it'll be upgraded to the base version (but no later).

At least this is my understanding - I couldn't find much documentation on the `--upgrade` flag.

Reply:
Okay, that would be the perfect behaviour. Could you confirm it locally with a small test case?

Reply:
Turns out it's not needed for packages from the base image. When upgraded, they are constrained with the newer version, so they will always be upgraded when running `pip-compile` in the service as well.

However, the `--upgrade` behaviour applies to service-specific packages. Without the flag, the previously installed version is kept, with it, they are also upgraded to the latest version. I think that is the behaviour we want to default to (upgrade everything).